Ruby Course (various): Link updates as per Issue 28290 (#28392)
TheOdinProject/curriculum: ruby/testing_ruby_with_rspec/project_connect_four.md

  1. Clone our [ruby testing repo](https://github.com/TheOdinProject/ruby_testing) and complete the lessons in the spec folder.
  1. Write tests for your [Tic Tac Toe project](/lessons/ruby-tic-tac-toe). In this situation, it's not quite as straightforward as just coming up with inputs and making sure the method returns the correct thing. You'll need to make sure the tests that determine victory or loss conditions are correctly assessed.
      1. Start by writing tests to make sure players win when they should, e.g. when the board reads X X X across the top row, your `#game_over` method (or its equivalent) should trigger.
      1. Test each of your critical methods to make sure they function properly and handle edge cases.
      1. Use mocks/doubles to isolate methods to make sure that they're sending back the right outputs.

</div>

### Project: TDD Connect Four

> Hopefully everyone has played Connect Four at some point (if not, see the Connect Four [Wikipedia page](http://en.wikipedia.org/wiki/Connect_Four)).  It's a basic game where each player takes turns dropping pieces into the cage.  Players win if they manage to get 4 of their pieces consecutively in a row, column, or along a diagonal.

Reviewer: Same here, "Wikipedia page - link" isn't sufficiently accessible and would force the user to have to have the whole line read out to try and get the context for what the link contains or leads to. In the suggestion, "Connect Four - link" is more descriptive because it leads directly to a page titled "Connect Four".

```suggestion
Hopefully most people will have played [Connect Four](http://en.wikipedia.org/wiki/Connect_Four) at some point. It's a basic game where each player takes turns dropping pieces into the cage. Players win if they manage to get 4 of their pieces consecutively in a row, column, or along a diagonal.
```

### Assignment 2

<div class="lesson-content__panel" markdown="1">

  1. Build Connect Four!  Just be sure to keep it TDD.

</div>

### Additional resources

This section contains helpful links to related content. It isn't required, so consider it supplemental.

- Stack Overflow has an [RSpec Mock Object example](http://stackoverflow.com/questions/3622604/rspec-mock-object-example).

> - Youtube has a [helpful video](https://www.youtube.com/watch?v=URSWYvyc42M) by Sandi Metz about Ruby testing.

Reviewer: ```suggestion
- Youtube has a [helpful video by Sandi Metz about Ruby testing](https://www.youtube.com/watch?v=URSWYvyc42M).
```
